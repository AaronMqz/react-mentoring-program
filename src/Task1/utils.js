export const GenresList = ["ALL", "DOCUMENTARY", "COMDEY", "HORROR", "CRIME"];
