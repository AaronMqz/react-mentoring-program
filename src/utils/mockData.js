export const mockData = {
  image_url: "https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg",
  movie_name: "Avengers: Inifinity War",
  release_year: "2018",
  genres: ["Action", "Adventure"],
  duration: "2h 29min",
  description:
    "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment, the fate of Earth and existence has never been more uncertain.",
  rating: "8.4",
};

export const SortByOptions = [
  { id: "release_date", name: "RELEASE DATE" },
  { id: "title", name: "TITLE" },
];
