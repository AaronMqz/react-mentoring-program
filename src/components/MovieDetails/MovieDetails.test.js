import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import DetailComponent from "./MovieDetails";
import { dummyMovieState } from "../../utils";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("MovieDetails Tests:", () => {
  test("1. That component renders movie details.", () => {
    render(<DetailComponent movie={dummyMovieState} />);
    const movieName = screen.getByText(dummyMovieState.movie_name);
    expect(movieName).toBeInTheDocument();
  });

  it("2. Snapshot renders correctly", () => {
    const tree = renderer
      .create(<DetailComponent movie={dummyMovieState} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
