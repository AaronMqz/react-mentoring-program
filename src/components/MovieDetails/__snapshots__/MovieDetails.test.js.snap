// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Tests: 2. Snapshot renders correctly 1`] = `
<div
  className="sc-beqWaB khSiD"
>
  <div
    className="sc-dmqHEX hRwmFB"
  >
    <img
      className="sc-hLseeU bMEDUl"
      src="https://m.media-amazon.com/images/I/71eHZFw+GlL._AC_SY879_.jpg"
    />
    <div
      className="sc-eDDNvR erzAVJ"
    >
      <div
        className="sc-iAEyYk fIvPjx"
      >
        <label
          aria-label="movie-name"
          className="sc-gueYoa fThXFl"
        >
          Avengers: Inifinity War
        </label>
        <div
          className="sc-jTrPJq fAsNMO"
        >
          8.4
        </div>
      </div>
      <div
        className="sc-gLDzan eYa-dv"
      >
        Action & Adventure
      </div>
      <div
        className="sc-iAEyYk fIvPjx"
      >
        <div
          className="sc-fsQiph ejcRNG"
        >
          2018
        </div>
        <div
          className="sc-fsQiph ejcRNG"
        >
          2h 29min
        </div>
      </div>
      <div
        className="sc-qRumB jkgUDs"
      >
        As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment, the fate of Earth and existence has never been more uncertain.
      </div>
    </div>
  </div>
</div>
`;
